src/python/gudhi/datasets/remote.py

```diff
@@ -0,0 +1,206 @@
+    chunk_size = 4096
+    with open(file_path,"rb") as f:
+        # Read and update hash string value in blocks of 4K
+        while True:
+            buffer = f.read(chunk_size)
+            if not buffer:
+                break
+            sha256_hash.update(buffer)
+    return sha256_hash.hexdigest()
+
+def _fetch_remote(url, filename, dirname, file_checksum = None, accept_license = False):
+    """
+    Fetch the wanted dataset from the given url and save it in file_path.
+
+    Parameters
+    ----------
+    url : string
+        The url to fetch the dataset from.
+    filename : string
+        The name to give to downloaded file.
+    dirname : string
+        The directory to save the file to.
+    file_checksum : string
+        The file checksum using sha256 to check against the one computed on the downloaded file.
+        Default is 'None', which means the checksum is not checked.
+    accept_license : boolean
+        Flag to specify if user accepts the file LICENSE and prevents from printing the corresponding license terms.
+        Default is False.
+
+    Returns
+    -------
+    file_path: string
+        Full path of the created file.
+
+    Raises
+    ------
+    IOError
+        If the computed SHA256 checksum of file does not match the one given by the user.
+    """
+
+    file_path = join(dirname, filename)
+
+    # Get the file
+    urlretrieve(url, file_path)
+
+    if file_checksum is not None:
+        checksum = _checksum_sha256(file_path)
+        if file_checksum != checksum:
+            # Remove file and raise error
+            remove(file_path)
+            raise IOError("{} has a SHA256 checksum : {}, "
+                        "different from expected : {}."
+                        "The file may be corrupted or the given url may be wrong !".format(file_path, checksum, file_checksum))
+
+    # Print license terms unless accept_license is set to True
+    if not accept_license:
+        license_file = join(dirname, "LICENSE")
+        if exists(license_file) and (file_path != license_file):
+            with open(license_file, 'r') as f:
+                print(f.read())
+
+    return file_path
+
+def _get_archive_and_dir(dirname, filename, label):
+    if dirname is None:
+        dirname = join(get_data_home(dirname), label)
+        makedirs(dirname, exist_ok=True)
+    else:
+        dirname = get_data_home(dirname)
+
+    archive_path = join(dirname, filename)
+
+    return archive_path, dirname
+
+def fetch_spiral_2d(filename = "spiral_2d.npy", dirname = None):
+    """
+    Fetch spiral_2d dataset remotely.
+
+    Parameters
+    ----------
+    filename : string
+        The name to give to downloaded file. Default is "spiral_2d.npy".
+    dirname : string
+        The directory to save the file to. Default is None, meaning that the downloaded file will be put in "~/gudhi_data/points/spiral_2d".
+
+    Returns
+    -------
+    points: numpy array
+        Array of shape (114562, 2).
```

Comment:
Another question: what should the dtype be? Currently, it is float64. Looking at the CSV file, all coordinates use only 6 (decimal) digits, so float32 looks good enough. On the other hand, more algorithms in gudhi seem to expect float64, so maybe it makes sense to provide data with float64.